backend/routes/auth.js

```diff
@@ -128,4 +137,121 @@ router.post(
   }
 );
 
+/**
+ * @method - POST
+ * @param - /forgot
+ * @description - Saves user's forgot password token in DB and sends user email
+ */
+router.post(
+  "/forgot",
+  [check("email", "Please enter a valid email").not().isEmpty().isEmail()],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({
+        message: errors.errors[0].msg,
+      });
+    }
+
+    const { email } = req.body;
+
+    try {
+      let user = await User.findOne({
+        email,
+      });
+
+      if (!user) {
+        return res.status(400).json({
+          message: "Invalid email! Please try again.",
+        });
+      }
+
+      const randomString = Math.random().toString(36).slice(2);
```

Comment:
This will probably work "well enough" for what we're doing with it; however, this could give you an empty string if `36 * Math.random()` happens to be an integer. Here's another way to generate a random string: <https://stackoverflow.com/a/27747377>

Reply:
Isn't that only the case if `Math.random()` is 0, which is an extremely low chance?

Reply:
Oh, you're right. Since it's slicing after the decimal point, it would have to be zero for there to be nothing after the decimal point.

However, the length of the token will still vary, depending on the specific random value that comes out. For example, there's a 1 in 36 chance that the last digit would be a 0, so it'd be omitted from the string representation (making the token length 9 instead of 10). I don't think that's a big problem either, but that can also be fixed with a while loop:

```javascript
let randomString = "";
while (randomString.length !== 10) {
  randomString = Math.random().toString(36).slice(2, 12);
}
```

In practice, I agree that this won't make a big difference either way. I'll approve the PR now, and you can change this if you feel like it.

Reply:
Yeah for sure, we noticed that as well. Definitely not a big deal for our purposes. Thank you!